### PR TITLE
First pass of route versioning.

### PR DESCRIPTION
**Pull request by kshakir**

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/vault/BossClientService.scala [hunk 1]
```diff
@@ -20,10 +20,10 @@ object BossClientService {
   case class BossResolveObject(obj: BossResolutionRequest, id: String, creationKey: String)
   case class BossObjectResolved(bossObject: BossResolutionResponse, creationKey: String)
 
-  def props(requestContext: RequestContext): Props = Props(new BossClientService(requestContext))
+  def props(requestContext: RequestContext, version: Int): Props = Props(new BossClientService(requestContext, version))
 }
 
-case class BossClientService(requestContext: RequestContext) extends Actor {
+case class BossClientService(requestContext: RequestContext, version: Int) extends Actor {
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/vault/BossClientService.scala]: I don't think it makes sense to pass these versions to BOSS and DM client actors.  The DM behavior required by v3 of UBam Ingest has no connection to what v3 of Analysis Describe needs.

Comment (kshakir) [inline src/main/scala/org/broadinstitute/dsde/vault/BossClientService.scala]: OK, I can remove the passed version from the BOSS & DM client actors.

**Comment by jmthibault79** [comment]
Looks good!  Just the one change and a rebase.

**Comment by kshakir** [comment]
Mind giving it another once over, just in case, then merging if everything looks good?

**Pull request merged**